lib/gosu/constants.rb

```diff
@@ -0,0 +1,332 @@
+    "GP_1_BUTTON_5",
+    "GP_1_BUTTON_6",
+    "GP_1_BUTTON_7",
+    "GP_1_BUTTON_8",
+    "GP_1_BUTTON_9",
+    "GP_1_BUTTON_10",
+    "GP_1_BUTTON_11",
+    "GP_1_BUTTON_12",
+    "GP_1_BUTTON_13",
+    "GP_1_BUTTON_14",
+    "GP_1_BUTTON_15",
+
+    "GP_2_DPAD_LEFT",
+    "GP_2_DPAD_RIGHT",
+    "GP_2_DPAD_UP",
+    "GP_2_DPAD_DOWN",
+    "GP_2_LEFT",
+    "GP_2_RIGHT",
+    "GP_2_UP",
+    "GP_2_DOWN",
+    "GP_2_BUTTON_0",
+    "GP_2_BUTTON_1",
+    "GP_2_BUTTON_2",
+    "GP_2_BUTTON_3",
+    "GP_2_BUTTON_4",
+    "GP_2_BUTTON_5",
+    "GP_2_BUTTON_6",
+    "GP_2_BUTTON_7",
+    "GP_2_BUTTON_8",
+    "GP_2_BUTTON_9",
+    "GP_2_BUTTON_10",
+    "GP_2_BUTTON_11",
+    "GP_2_BUTTON_12",
+    "GP_2_BUTTON_13",
+    "GP_2_BUTTON_14",
+    "GP_2_BUTTON_15",
+
+    "GP_3_DPAD_LEFT",
+    "GP_3_DPAD_RIGHT",
+    "GP_3_DPAD_UP",
+    "GP_3_DPAD_DOWN",
+    "GP_3_LEFT",
+    "GP_3_RIGHT",
+    "GP_3_UP",
+    "GP_3_DOWN",
+    "GP_3_BUTTON_0",
+    "GP_3_BUTTON_1",
+    "GP_3_BUTTON_2",
+    "GP_3_BUTTON_3",
+    "GP_3_BUTTON_4",
+    "GP_3_BUTTON_5",
+    "GP_3_BUTTON_6",
+    "GP_3_BUTTON_7",
+    "GP_3_BUTTON_8",
+    "GP_3_BUTTON_9",
+    "GP_3_BUTTON_10",
+    "GP_3_BUTTON_11",
+    "GP_3_BUTTON_12",
+    "GP_3_BUTTON_13",
+    "GP_3_BUTTON_14",
+    "GP_3_BUTTON_15",
+
+    "GP_LEFT_STICK_X_AXIS",
+    "GP_LEFT_STICK_Y_AXIS",
+    "GP_RIGHT_STICK_X_AXIS",
+    "GP_RIGHT_STICK_Y_AXIS",
+    "GP_LEFT_TRIGGER_AXIS",
+    "GP_RIGHT_TRIGGER_AXIS",
+
+    "GP_0_LEFT_STICK_X_AXIS",
+    "GP_0_LEFT_STICK_Y_AXIS",
+    "GP_0_RIGHT_STICK_X_AXIS",
+    "GP_0_RIGHT_STICK_Y_AXIS",
+    "GP_0_LEFT_TRIGGER_AXIS",
+    "GP_0_RIGHT_TRIGGER_AXIS",
+
+    "GP_1_LEFT_STICK_X_AXIS",
+    "GP_1_LEFT_STICK_Y_AXIS",
+    "GP_1_RIGHT_STICK_X_AXIS",
+    "GP_1_RIGHT_STICK_Y_AXIS",
+    "GP_1_LEFT_TRIGGER_AXIS",
+    "GP_1_RIGHT_TRIGGER_AXIS",
+
+    "GP_2_LEFT_STICK_X_AXIS",
+    "GP_2_LEFT_STICK_Y_AXIS",
+    "GP_2_RIGHT_STICK_X_AXIS",
+    "GP_2_RIGHT_STICK_Y_AXIS",
+    "GP_2_LEFT_TRIGGER_AXIS",
+    "GP_2_RIGHT_TRIGGER_AXIS",
+
+    "GP_3_LEFT_STICK_X_AXIS",
+    "GP_3_LEFT_STICK_Y_AXIS",
+    "GP_3_RIGHT_STICK_X_AXIS",
+    "GP_3_RIGHT_STICK_Y_AXIS",
+    "GP_3_LEFT_TRIGGER_AXIS",
+    "GP_3_RIGHT_TRIGGER_AXIS",
+  ]
+
+  constants = constants + [
+    "MAJOR_VERSION",
+    "MINOR_VERSION",
+    "POINT_VERSION",
+  ]
+
+  constants = constants + [
+    "WF_WINDOWED",
+    "WF_FULLSCREEN",
+    "WF_RESIZABLE",
+    "WF_BORDERLESS",
+
+    "BM_DEFAULT",
+    "BM_INTERPOLATE",
+    "BM_ADD",
+    "BM_MULTIPLY",
+
+    "FF_BOLD",
+    "FF_ITALIC",
+    "FF_UNDERLINE",
+    "FF_COMBINATIONS",
+
+    "AL_LEFT",
+    "AL_RIGHT",
+    "AL_CENTER",
+    "AL_JUSTIFY",
+
+    "IF_SMOOTH",
+    "IF_TILEABLE_LEFT",
+    "IF_TILEABLE_TOP",
+    "IF_TILEABLE_RIGHT",
+    "IF_TILEABLE_BOTTOM",
+    "IF_TILEABLE",
+    "IF_RETRO",
+  ]
+
+  constants.each do |const|
+    attach_variable :"#{const}", :"Gosu_#{const}", :uint32
+    Gosu.const_set(const, Gosu.send(:"#{const}"))
```

Comment:
You don't need conversions if you already fill in your `constants` Array with Symbols instead of Strings:

```suggestion
  constants.each do |const|
    attach_variable const, :"Gosu_#{const}", :uint32
    Gosu.const_set(const, Gosu.send(const))
```